### PR TITLE
Replace webpack with esbuild

### PR DESCRIPTION
**Pull request by maxkatz6**
Continues https://github.com/AvaloniaUI/Avalonia/pull/8820
1. Replaces webpack with esbuild to simplify build process and speed
2. Lowers down target from es2020 to es2016 (+AsyncIterator from 2018)
3. For the previewer es6 is used (instead of es5)

Tested previewer locally with 
`
dotnet exec --runtimeconfig .\Sandbox.runtimeconfig.json --depsfile .\Sandbox.deps.json C:/Users/maxka/source/repos/Avalonia/src/tools/Avalonia.Designer.HostApp/bin/Debug/netcoreapp2.0/Avalonia.Designer.HostApp.dll --transport file:///C:/Users/maxka/source/repos/Avalonia/samples/Sandbox/MainWindow.axaml --method html --html-url http://127.0.0.1:6001 .\Sandbox.dll
`

### CODE REVIEW

File: src/Web/Avalonia.Web.Blazor/Avalonia.Web.Blazor.csproj [hunk 1]
```diff
@@ -31,7 +31,6 @@
 
   <ItemGroup>
     <PackageReference Include="Microsoft.AspNetCore.Components.Web" Version="6.0.8" />
-    <PackageReference Include="Microsoft.TypeScript.MSBuild" Version="4.7.4" PrivateAssets="all" />
```
Comment (maxkatz6) [inline src/Web/Avalonia.Web.Blazor/Avalonia.Web.Blazor.csproj:34]: It shouldn't be here. We don't compile ts with msbuild anymore.

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0023747-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0023757-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0023760-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

**Review by danwalmsley — APPROVED** [review]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0023851-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

**Pull request merged**